### PR TITLE
allow to read password from file

### PR DESCRIPTION
**Pull request by Kielek**
it enables option to utilize docker swarm secret storage for password instead of env variables or app switches

### CODE REVIEW

**Comment by Kielek** [comment]
@bboreham, if you are fine with these changes, it will be great to have new _offical release_. 1.13.3?

File: site/faq.md [hunk 1]
```diff
@@ -33,10 +33,14 @@ OSS Scope has no user concept, this is only available in Weave Cloud. To limit t
         Password for basic authentication (default "admin")
   --app.basicAuth.username string
         Username for basic authentication (default "admin")
+  --probe.basicAuth.password.filename string
+        Password filename for basic authentication. It overwrites app.basicAuth.password
```
Comment (bboreham) [inline site/faq.md:37]: Is this meant to be `--app...` ?

File: site/faq.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ OSS Scope has no user concept, this is only available in Weave Cloud. To limit t
 
 - setup a reverse proxy with auth and block access to non admin users,
 - capture the calls with something like Chrome network console to get the endpoints to know which requests to authenticate in the proxy server.
-- you can use Basic HTTP Auth since Scope 1.10.0 - just use these command line
+- you can use Basic HTTP Auth since Scope 1.10.0 (support for loading password from file sice Scope 1.13.3) - just use these command line
```
Comment (bboreham) [inline site/faq.md:26]: Typo s/sice/since/

Put this in a separate PR so we don't publish incorrect info for a time.

File: prog/main.go [hunk 3]
```diff
@@ -504,3 +518,15 @@ func main() {
...
 }
+
+func getPassword(password string, passwordFilename string) string {
+	if passwordFilename == "" {
+		return password
+	}
+
+	b, err := ioutil.ReadFile(passwordFilename)
+	if err != nil {
+		log.Errorf("Cannor read password from: %v", passwordFilename)
```
Comment (bboreham) [inline prog/main.go:529]: Typo s/Cannor/Cannot/

Comment (bboreham) [inline prog/main.go:529]: Please include `err` in the message.

**Review by bboreham — CHANGES_REQUESTED** [review]
Thanks for the PR.  The concept is fine; I made some small points inline.

Bigger point though: what happens if the file is not readable?  Looks to me like the system will then be open to all callers, which is probably not what the administrator would expect if they configured a password file.

File: prog/main.go (hunk 3 above)
Comment (Kielek) [reply prog/main.go]: fixed

Comment (Kielek) [reply prog/main.go]: fixed

File: site/faq.md (hunk 2 above)
Comment (Kielek) [reply site/faq.md]: fixed, faq moved to #3852

File: site/faq.md (hunk 1 above)
Comment (Kielek) [reply site/faq.md]: fixed, faq moved to #3852

**Comment by Kielek** [comment]
> Bigger point though: what happens if the file is not readable? Looks to me like the system will then be open to all callers, which is probably not what the administrator would expect if they configured a password file.

You are right. I have change log to Fatal, it should close the process and I think, it is expected solution.

**Comment by Kielek** [comment]
@bboreham, any chance to move forward with this PR (and prepare next official release)?

**Comment by Kielek** [comment]
@bboreham, is there any chance that you check this PR?

**Review by bboreham — APPROVED** [review]
Thanks! Lgtm; just checking if integration tests pass
